### PR TITLE
Easy cookie

### LINKED ISSUE #71: inject cookie
## 🚀 Feature Proposal

Add the feature to send a request with cookies and read the cookies in the response easily

## Motivation

Simplify the cookie usage

## Example

```js
inject(dispatch, { method: 'get', url: '/', cookies: {
 foo: 'bar'
} }, (err, res) => {
  console.log(res.payload)
  console.log(res.cookies()) // returned by the server (same usage of .json() feature)
})
```

### PR DESCRIPTION
**Pull request by Eomm**
closes #71

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -50,6 +52,11 @@ function Request (options) {
   this.headers['user-agent'] = this.headers['user-agent'] || 'lightMyRequest'
   this.headers.host = this.headers.host || options.authority || hostHeaderFromURL(parsedURL)
 
+  if (options.cookies) {
+    const { cookies } = options
+    this.headers.cookie = Object.keys(cookies).map(key => cookie.serialize(key, cookies[key])).join('; ')
```
Comment (mcollina) [inline lib/request.js]: I think we should document the fact that the cookies option take precedence over the cookie header.

File: lib/response.js [hunk 2]
```diff
@@ -127,6 +129,10 @@ function generatePayload (response) {
     return JSON.parse(this.payload)
   }
 
+  res.cookies = function parseCookies () {
+    return setCookie.parse(this)
+  }
```
Comment (mcollina) [inline lib/response.js]: I would make this a getter instead. I would also create a new class that inherits from ServerResponse with it. (use util.inherits, not class).

File: lib/request.js (hunk 1 above)
Comment (Eomm) [reply lib/request.js]: Maybe we could just append, so the user will decide

File: lib/response.js (hunk 2 above)
Comment (Eomm) [reply lib/response.js]: Is not the inherit already done here?

https://github.com/fastify/light-my-request/blob/7acd5444048b1bad60bca6edd163401b51e0181f/lib/response.js#L47

Gotcha for the getter

**Comment by Eomm** [comment]
Good point, but I found [this messge](https://github.com/jshttp/cookie/issues/58#issuecomment-230946655) by the creator of `cookie` that says that the module's parse function is only designed to parse the Cookie header, not the Set-Cookie header.

**Comment by mcollina** [comment]
@Eomm would you mind to send a PR to update the fastify docs?

**Pull request merged**